fedscale_examples/Async_FL/resource_manager.py

```diff
@@ -0,0 +1,47 @@
+from fedscale.core import events
+import threading
+
+
+class ResourceManager(object):
+    """Schedule training tasks across GPUs/CPUs"""
+
+    def __init__(self, experiment_mode):
+
+        self.client_run_queue = []
+        self.experiment_mode = experiment_mode
+        self.update_lock = threading.Lock()
+
+    def get_remaining(self):
+        return len(self.client_run_queue)
+
+    def register_tasks(self, clientsToRun):
+        self.client_run_queue += clientsToRun.copy()
+
+    def remove_client_task(self, client_id):
+        assert (client_id in self.client_run_queue,
+                f"client task {client_id} is not in task queue")
+        pass
+
+    def has_next_task(self, client_id=None):
+        exist_next_task = False
+        if self.experiment_mode == events.SIMULATION_MODE:
+            exist_next_task = len(self.client_run_queue) > 0
+        else:
+            exist_next_task = client_id in self.client_run_queue
+        return exist_next_task
+
+    def get_next_task(self, client_id=None):
+        next_task_id = None
+        self.update_lock.acquire()
+        if self.experiment_mode == events.SIMULATION_MODE:
+            if self.has_next_task(client_id):
+                next_task_id = self.client_run_queue[0]
+                self.client_run_queue.pop(0)
+        else:
+            if client_id in self.client_run_queue:
+                next_task_id = client_id
+                self.client_run_queue.remove(next_task_id)
+
+        self.update_lock.release()
+        print(f"@@@@@@ Schedule next client {next_task_id}")
```

Comment:
Please do not use print. #101 . Also we need to avoid using random formats (e.g., "@@@@").

Reply:
forget to remove😅